spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTemplate.java

```diff
@@ -228,65 +228,68 @@ public long executeDmlStatement(Statement statement) {
 
 	@Override
 	public void insert(Object object) {
-		applySaveMutations(this.mutationFactory.insert(object), Collections.singletonList(object), null);
```

Comment:
Since objects are passed by reference, the fact that `insert` modifies them shouldn't be a problem.

Reply:
It's not that the entity objects are modified, but that the mutations generated from them would be out of date (they're created new from the entity objects for the save operation).